### ISSUE TITLE
docker exec does not wait on paused container

### ISSUE THREAD

**Issue by michaelkebe** [docker/docker#9672]
`man docker-exec` says: 

> If the container is paused, then the docker exec command will wait until the container is unpaused, and then run

But I get an error:

> ```FATA[0000] Error response from daemon: Container distracted_lumiere is paused, unpause the container before exec```

How to get the error: 
```
$ docker run -d mongo:latest 
9696a9b2d5726339a30c265e2587a11cd8fcb8f31a2419f9081583b3fb87b120
$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS               NAMES
9696a9b2d572        mongo:latest        "/entrypoint.sh mong   8 seconds ago       Up 7 seconds        27017/tcp           distracted_lumiere   
$ docker pause distracted_lumiere 
distracted_lumiere
$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS                   PORTS               NAMES
9696a9b2d572        mongo:latest        "/entrypoint.sh mong   33 seconds ago      Up 32 seconds (Paused)   27017/tcp           distracted_lumiere   
$ docker exec 
9696a9b2d572        distracted_lumiere  
$ docker exec distracted_lumiere echo foo 
FATA[0000] Error response from daemon: Container distracted_lumiere is paused, unpause the container before exec 
```

Ubuntu 14.04
```
$ uname -a
Linux XXXXXXXX 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
$ docker version
Client version: 1.4.0
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 4595d4f
OS/Arch (client): linux/amd64
Server version: 1.4.0
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 4595d4f
$ docker -D info
Containers: 1
Images: 53
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Dirs: 55
Execution Driver: native-0.2
Kernel Version: 3.13.0-43-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 4
Total Memory: 7.506 GiB
Name: XXXXXXXXXX
ID: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
Debug mode (server): false
Debug mode (client): true
Fds: 17
Goroutines: 16
EventsListeners: 0
Init Path: /usr/bin/docker
Docker Root Dir: /var/lib/docker
WARNING: No swap limit support
```

**Comment by crosbymichael** [comment]
@SvenDowideit The man pages are wrong, they should be changed as the implementation is the correct way to handle this situation.

**Comment by SvenDowideit** [comment]
this was fixed in #9356 and has been published.
